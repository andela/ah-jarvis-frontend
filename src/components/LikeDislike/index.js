import React from 'react';
import PropTypes from 'prop-types';

const Reaction = ({
  id, src, count, onClick,
}) => (
  <div className="reactions__icon p-b--20">
    <span>{count}</span>
    <a href="#!" onClick={onClick} className="">
      <img id={id} src={src} alt="Reaction" className="" />
    </a>
  </div>
);

export default Reaction;

Reaction.propTypes = {
  id: PropTypes.any.isRequired,
  src: PropTypes.string.isRequired,
  count: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
};
