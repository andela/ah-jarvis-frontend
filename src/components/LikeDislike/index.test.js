import { shallow } from 'enzyme';
import React from 'react';
import thumbsUp from '../../assets/icons/thumbsUp.svg';
import LikeDislike from '.';

describe('<LikeDislike />', () => {
  it('renders  <LikeDislike /> components', () => {
    const snap = shallow(<LikeDislike id="" src={thumbsUp} count="" onClick={() => {}} />);
    expect(snap).toMatchSnapshot();
  });
});
