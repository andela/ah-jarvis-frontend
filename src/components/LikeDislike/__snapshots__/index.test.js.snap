// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LikeDislike /> renders  <LikeDislike /> components 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Reaction
    count=""
    id=""
    onClick={[Function]}
    src="thumbsUp.svg"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span />,
        <a
          className=""
          href="#!"
          onClick={[Function]}
>
          <img
                    alt="Reaction"
                    className=""
                    id=""
                    src="thumbsUp.svg"
          />
</a>,
      ],
      "className": "reactions__icon p-b--20",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Reaction"
            className=""
            id=""
            src="thumbsUp.svg"
/>,
          "className": "",
          "href": "#!",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Reaction",
            "className": "",
            "id": "",
            "src": "thumbsUp.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span />,
          <a
            className=""
            href="#!"
            onClick={[Function]}
>
            <img
                        alt="Reaction"
                        className=""
                        id=""
                        src="thumbsUp.svg"
            />
</a>,
        ],
        "className": "reactions__icon p-b--20",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Reaction"
              className=""
              id=""
              src="thumbsUp.svg"
/>,
            "className": "",
            "href": "#!",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Reaction",
              "className": "",
              "id": "",
              "src": "thumbsUp.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
